base/array.jl

```diff
@@ -2502,6 +2502,29 @@ function filter!(f, a::AbstractVector)
     return a
 end
 
+"""
+    filter(f)
+
+Create a function that filters its arguments with function `f` using [`filter`](@ref), i.e.
+a function equivalent to `x -> filter(f, x)`.
+
+The returned function is of type `Base.Fix1{typeof(filter)}`, which can be
+used to implement specialized methods.
+
+# Example usage:
+```jldoctest
+julia> -5:5 |> filter(<(-3))
+2-element Vector{Int64}:
+ -5
+ -4
```

Comment:
I presume that the motivation for this is to pass it to higher-order functions, so perhaps the examples should show that? 
```suggestion
julia> (1, 2, NaN, 4) |> filter(!isnan)
(1, 2, 4)

julia> map(filter(iseven), [1:3, 2:4, 3:5])
3-element Vector{Vector{Int64}}:
 [2]
 [2, 4]
 [4]
```